src/uwtools/tests/config/formats/test_base.py

```diff
@@ -0,0 +1,149 @@
+def test_compare_config(caplog, fmt, salad_base):
+    """
+    Compare two config objects.
+    """
+    log.setLevel(logging.INFO)
+    cfgobj = tools.format_to_config(fmt)(fixture_path(f"simple.{fmt}"))
+    if fmt == FORMAT.ini:
+        salad_base["salad"]["how_many"] = "12"  # str "12" (not int 12) for ini
+    assert cfgobj.compare_config(salad_base) is True
+    # Expect no differences:
+    assert not caplog.records
+    caplog.clear()
+    # Create differences in base dict:
+    salad_base["salad"]["dressing"] = "italian"
+    salad_base["salad"]["size"] = "large"
+    del salad_base["salad"]["how_many"]
+    assert not cfgobj.compare_config(cfgobj, salad_base)
+    assert not cfgobj.compare_config(salad_base)
+    # Expect to see the following differences logged:
+    for msg in [
+        "salad:        how_many:  - 12 + None",
+        "salad:        dressing:  - balsamic + italian",
+        "salad:            size:  - None + large",
+    ]:
+        assert logged(caplog, msg)
+
+
+def test_depth(config):
+    assert config.depth == 1
+
+
+def test_dereference(caplog, config):
+    # Test demonstrates that:
+    #   - Config dereferencing uses environment variables.
+    #   - Initially-unrenderable values do not cause errors.
+    #   - Initially-unrenderable values may be rendered via iteration.
+    #   - Finally-unrenderable values do not cause errors and are returned unmodified.
+    log.setLevel(logging.DEBUG)
+    config.data.update({"a": "{{ b.c + 11 }}", "b": {"c": "{{ N | int + 11 }}"}, "d": "{{ X }}"})
+    with patch.dict(os.environ, {"N": "55"}, clear=True):
+        config.dereference()
+    for excerpt in [
+        "'a': '{{ b.c + 11 }}', 'b': {'c': '{{ N | int + 11 }}'}, 'd': '{{ X }}'",
+        "'a': '{{ b.c + 11 }}', 'b': {'c': 66}, 'd': '{{ X }}'",
+        "'a': 77, 'b': {'c': 66}, 'd': '{{ X }}'",
+    ]:
+        assert regex_logged(caplog, excerpt)
+    assert config == {"foo": 88, "a": 77, "b": {"c": 66}, "d": "{{ X }}"}
+
+
+# NB: Need direct test for parse_include().
+
+# NB: Need direct test for update_values().
+
+
+@pytest.mark.parametrize("fmt1", [FORMAT.ini, FORMAT.nml, FORMAT.yaml])
+@pytest.mark.parametrize("fmt2", [FORMAT.ini, FORMAT.nml, FORMAT.yaml])
+def test_transform_config(fmt1, fmt2, tmp_path):
+    """
+    Test that transforms config objects to objects of other config subclasses.
+    """
+    outfile = tmp_path / f"test_{fmt1.lower()}to{fmt2.lower()}_dump.{fmt2}"
+    reference = fixture_path(f"simple.{fmt2}")
+    cfgin = tools.format_to_config(fmt1)(fixture_path(f"simple.{fmt1}"))
+    tools.format_to_config(fmt2).dump_dict(path=outfile, cfg=cfgin.data)
+    with open(reference, "r", encoding="utf-8") as f1:
+        reflines = [line.strip().replace("'", "") for line in f1]
+    with open(outfile, "r", encoding="utf-8") as f2:
+        outlines = [line.strip().replace("'", "") for line in f2]
+    for line1, line2 in zip(reflines, outlines):
+        assert line1 == line2
```

Comment:
This is a funny (but valuable) test that doesn't seem to belong anywhere: It tests the interaction of various formats and so doesn't really belong to the test modules for any of those individual formats. I've put it here in `test_base.py`, which seems reasonable to me.